test/libsolidity/smtCheckerTests/operators/userDefined/user_defined_operations_on_constants_fail.sol

```diff
@@ -0,0 +1,104 @@
+function noteq(I16 x, I16 y) pure returns (bool) { return I16.unwrap(x) != I16.unwrap(y); }
+function lt(I16 x, I16 y) pure returns (bool) { return I16.unwrap(x) < I16.unwrap(y); }
+function gt(I16 x, I16 y) pure returns (bool) { return I16.unwrap(x) > I16.unwrap(y); }
+function leq(I16 x, I16 y) pure returns (bool) { return I16.unwrap(x) <= I16.unwrap(y); }
+function geq(I16 x, I16 y) pure returns (bool) { return I16.unwrap(x) >= I16.unwrap(y); }
+
+contract C {
+    I16 constant MINUS_TWO = I16.wrap(-2);
+    I16 constant ZERO = I16.wrap(0);
+    I16 constant ONE = I16.wrap(1);
+    I16 constant TWO = I16.wrap(2);
+    I16 constant THREE = I16.wrap(3);
+    I16 constant FOUR = I16.wrap(4);
+
+    function testBitwise() public pure {
+        assert(ONE | TWO == FOUR); // should fail
+        assert(ONE & THREE == FOUR); // should fail
+        assert(TWO ^ TWO == FOUR); // should fail
+        assert(~ONE == FOUR); // should fail
+    }
+
+    function testArithmetic() public pure {
+        assert(TWO + THREE == FOUR); // should fail
+        assert(TWO - TWO == FOUR); // should fail
+        assert(-TWO == FOUR); // should fail
+        assert(TWO * THREE == FOUR); // should fail
+        assert(TWO / TWO == FOUR); // should fail
+        assert(TWO % TWO == FOUR); // should fail
+    }
+
+    function testComparison() public pure {
+        assert(!(TWO == TWO)); // should fail
+        assert(TWO != TWO); // should fail
+        assert(TWO < TWO); // should fail
+        assert(TWO > TWO); // should fail
+        assert(!(TWO <= TWO)); // should fail
+        assert(!(TWO >= TWO)); // should fail
+    }
+}
+// ====
+// SMTEngine: all
+// ----
+// Warning 6756: (2012-2021): User-defined operators are not yet supported by SMTChecker. This invocation of operator | has been ignored, which may lead to incorrect results.
+// Warning 6756: (2012-2029): User-defined operators are not yet supported by SMTChecker. This invocation of operator == has been ignored, which may lead to incorrect results.
+// Warning 6756: (2062-2073): User-defined operators are not yet supported by SMTChecker. This invocation of operator & has been ignored, which may lead to incorrect results.
+// Warning 6756: (2062-2081): User-defined operators are not yet supported by SMTChecker. This invocation of operator == has been ignored, which may lead to incorrect results.
+// Warning 6756: (2114-2123): User-defined operators are not yet supported by SMTChecker. This invocation of operator ^ has been ignored, which may lead to incorrect results.
+// Warning 6756: (2114-2131): User-defined operators are not yet supported by SMTChecker. This invocation of operator == has been ignored, which may lead to incorrect results.
+// Warning 6156: (2164-2168): User-defined operators are not yet supported by SMTChecker. This invocation of operator ~ has been ignored, which may lead to incorrect results.
```

Comment:
Note that there are two different error codes for this (oddly, `error_codes.py` gave them misleadingly similar codes - WTF?). From tooling perspective I think it's not great to have multiple codes for the same thing but our current error check script is a bit annoying about this. It won't let you use the same code twice even if you know it's the same thing.

I considered extracting the piece reporting the warning into a helper but maybe it's an overkill for an error that will probably won't live longer than one release and won't be seen too often...

Reply:
Yeah, it's fine - indeed a weird coincidence though :-).